### PR TITLE
Fix session nonce re-use

### PR DESCRIPTION
**Pull request by bobf**
Use a new secure-random nonce for each session encryption.

Many thanks to @Trundle for writing [this gist](https://gist.github.com/Trundle/76c0d8f80999617f4d8373c03f86391a) highlighting the severity of this issue.

### CODE REVIEW

File: src/jetzig/Environment.zig [hunk 1]
```diff
@@ -95,8 +95,7 @@ pub fn getServerOptions(self: Environment) !jetzig.http.Server.ServerOptions {
         std.process.exit(1);
     }
 
-    // TODO: Generate nonce per session - do research to confirm correct best practice.
-    const secret_len = jetzig.http.Session.Cipher.key_length + jetzig.http.Session.Cipher.nonce_length;
+    const secret_len = jetzig.http.Session.Cipher.key_length;
     const secret = (try self.getSecret(&logger, secret_len, environment))[0..secret_len];
```
Comment (Trundle) [inline src/jetzig/Environment.zig:99]: The slice could now even be omitted, couldn't it? Unless it's a feature that too long values are allowed.

File: src/jetzig/http/Session.zig [hunk 2]
```diff
@@ -91,97 +75,70 @@ fn save(self: *Self) !void {
...
     defer self.allocator.free(encoded);
-
-    if (self.cookie) |*ptr| self.allocator.free(ptr.*.value);
-    self.cookie = .{ .value = try self.allocator.dupe(u8, encoded) };
-
-    try self.cookies.put(
-        cookie_name,
-        self.cookie.?,
-    );
+    std.debug.print("encoded: {s}\n", .{encoded});
```
Comment (Trundle) [inline src/jetzig/http/Session.zig:83]: Debug print

File: src/jetzig/http/Session.zig [hunk 3]
```diff
@@ -91,97 +75,70 @@ fn save(self: *Self) !void {
...
-        );
-        return buf[0..];
-    } else {
-        return self.allocator.dupe(u8, "hello");
-    }
+fn decrypt(self: *Self, data: []u8) ![]u8 {
+    const secret_bytes = std.mem.sliceAsBytes(self.encryption_key);
+    const key = secret_bytes[0..Cipher.key_length];
+    const nonce = data[0..Cipher.nonce_length];
+    const buf = try self.allocator.alloc(u8, data.len - Cipher.tag_length - Cipher.nonce_length);
```
Comment (Trundle) [inline src/jetzig/http/Session.zig:107]: The passed size potentially underflows if `data` is too short

File: src/jetzig/http/Session.zig [hunk 4]
```diff
@@ -91,97 +75,70 @@ fn save(self: *Self) !void {
...
-            key,
-        );
-        return buf[0..];
-    } else {
-        return self.allocator.dupe(u8, "hello");
-    }
+fn decrypt(self: *Self, data: []u8) ![]u8 {
+    const secret_bytes = std.mem.sliceAsBytes(self.encryption_key);
+    const key = secret_bytes[0..Cipher.key_length];
+    const nonce = data[0..Cipher.nonce_length];
```
Comment (Trundle) [inline src/jetzig/http/Session.zig:106]: Does this panic when `data` is too short?

File: src/jetzig/Environment.zig (hunk 1 above)
Comment (bobf) [reply src/jetzig/Environment.zig:99]: Yep, the secret key generator in the CLI tool over-allocates to allow for changes in backend crypto implemntation (such as this one).

File: src/jetzig/http/Session.zig (hunk 4 above)
Comment (bobf) [reply src/jetzig/http/Session.zig:106]: Yeah, but that's annoying - I'll add some runtime bounds checks to make it more friendly.

File: src/jetzig/http/Session.zig (hunk 3 above)
Comment (bobf) [reply src/jetzig/http/Session.zig:107]: Will fix with bounds checks - thanks !

**Pull request merged**